tractor/_spawn.py

```diff
@@ -18,34 +25,62 @@
 
 from . import _forkserver_override
 from ._state import current_actor
-from ._actor import Actor
+from .log import get_logger
+from ._portal import Portal
+from ._actor import Actor, ActorFailure
+
+
+log = get_logger('tractor')
+
+# use trip as our default for now
+if platform.system() != 'Windows':
+    _spawn_method: str = "trio_run_in_process"
+else:
+    _spawn_method = "spawn"
+
+_ctx: Optional[mp.context.BaseContext] = None
+
 
+if platform.system() == 'Windows':
+    async def proc_waiter(proc: mp.Process) -> None:
+        await trio.hazmat.WaitForSingleObject(proc.sentinel)
+else:
+    import trio_run_in_process
 
-_ctx: mp.context.BaseContext = mp.get_context("spawn")  # type: ignore
+    async def proc_waiter(proc: mp.Process) -> None:
+        await trio.hazmat.wait_readable(proc.sentinel)
 
 
-def try_set_start_method(name: str) -> mp.context.BaseContext:
+def try_set_start_method(name: str) -> Optional[mp.context.BaseContext]:
```

Comment:
ditto with above. Optional and Union return values is a code smell anyways

Reply:
The thing is for `trip` there is no context possible to return, nor is this function called in that case.
I'm not sure how else to specify the output if it's not this?

Reply:
Either make a wrapper function that does produce a context or this function simply decides on the start method given the platform yadayada and then outputs a string/enum. I don't think it blocks this PR btw

Reply:
KK we can discuss further in  #101